scripts/lib.sh

```diff
@@ -52,3 +52,26 @@ function get_finality_delay() {
 function sat_to_btc() {
     echo "scale=8; $1/100000000" | bc | awk '{printf "%.8f\n", $0}'
 }
+
+function sent_and_proof_btc() {
+  #$1 = PEG_IN_AMOUNT
+  #$2 = USE_GATEWAY
+  # Get a peg-in address, which is derived from the federation's descriptor in which every key was tweaked with the same
+  # random value only known to our client.
+  if [ "$2" == 1 ]; then ADDR="$($FM_LN1 -H gw-address)"; else ADDR="$($FM_MINT_CLIENT peg-in-address)"; fi
+
+  # We send the amount we want to peg-in to this address
+  TX_ID="$($FM_BTC_CLIENT sendtoaddress $ADDR "$(sat_to_btc $1)")"
+
+  # Now we "wait" for confirmations
+  $FM_BTC_CLIENT generatetoaddress 11 "$($FM_BTC_CLIENT getnewaddress)"
+  await_block_sync
+
+  # We then get a proof from our bitcoind that we sent coins to the peg-in address. This proof can be evaluated by just
+  # looking at block headers. The federation uses this so that it only needs to be aware of valid block hashes and not
+  # entire blocks.
+  TXOUT_PROOF="$($FM_BTC_CLIENT gettxoutproof "[\"$TX_ID\"]")"
+  TRANSACTION="$($FM_BTC_CLIENT getrawtransaction $TX_ID)"
+  export TXOUT_PROOF=$TXOUT_PROOF;
+  export TRANSACTION=$TRANSACTION;
```

Comment:
Also wondering if this return could be done more elegantly …

Reply:
I don't know how. `echo` them with a line break would be weird. Setting a global variable like `proof_and_transaction=(a,b)` or `proof=..; transaction=..` ?

Reply:
Maybe return them as json and use `jq` to read from it? Idk, what do other people think? @justinmoon @jkitman